viewing_party/party.py

```diff
@@ -0,0 +1,226 @@
+    friends_watched = []
+
+    for movie in user_data["watched"]:
+        movies_watched.append(movie["title"])
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friends_watched.append(movie["title"])
+
+    #compare movies_watched with friends_watched
+    #.remove method on movies_watched
+
+    for movie in friends_watched:
+        if movie in movies_watched:
+            movies_watched.remove(movie)
+
+    # unique_movies = []
+    # for movie in movies_watched:
+    #     unique_movies.append({"title": movie})
+
+    converted_movies_watched = set(movies_watched)
+    converted_friends_watched = set(friends_watched)
+
+    result_converted = converted_movies_watched - converted_friends_watched
+
+    unique_movies = []
+    for movie in result_converted:
+        unique_movies.append({"title": movie})
+    
+    return unique_movies
+
+def get_friends_unique_watched(user_data):
+    movies_watched = []
+    friends_watched = []
+
+    for movie in user_data["watched"]:
+        movies_watched.append(movie["title"])
+    
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friends_watched.append(movie["title"])
+    
+    for movie in friends_watched:
+        if movie in movies_watched:
+            friends_watched.remove(movie)
+
+    converted_movies_watched = set(movies_watched)
+    converted_friends_watched = set(friends_watched)
+
+    result_converted = converted_friends_watched - converted_movies_watched
+
+    unique_movies_friends = []
+    for movie in result_converted:
+        unique_movies_friends.append({"title": movie})
+    
+    return unique_movies_friends
+
+def get_available_recs(user_data):
+    movies_watched = []
+    friends_watched = []
+    movies_recommended = []
+
+    for movie in user_data["watched"]:
+        movies_watched.append(movie)
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friends_watched.append(movie)
+
+    for user_movie in movies_watched:
+        found = []
+        for friend_movie in friends_watched:
+            if friend_movie["title"] == user_movie["title"]:
+                found.append(friend_movie)
+        
+        for found_movie in found:
+            friends_watched.remove(found_movie)
+        # if movie in movies_watched:
+        #     friends_watched.remove(movie)
+    
+    for friend_movie in friends_watched:
+        if friend_movie["host"] in user_data["subscriptions"] and friend_movie not in movies_recommended:
+            movies_recommended.append(friend_movie)
+    
+    return movies_recommended
+
+    
+    
+
+
+    #iterate through what friends have watched
+    # create a list that describes the movies for recommend
+    # if movie title is in watched and host is in subscriptions, then add it to movies_recommended
+    # return a list of dictionaries 
+
+def get_new_rec_by_genre(user_data):
+    movies_watched = []
```

Comment:
This function would have been a great place to use the `get_friends_unique_watched()` and `get_most_watched_genre()`.